samples/Clients/src/BlazorAuth/BlazorAuth/Pages/_Host.cshtml.cs

```diff
@@ -0,0 +1,32 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace BlazorAuth.Pages
+{
+    public class HostModel : PageModel
+    {
+        [BindProperty(SupportsGet = true)] public string RedirectUri { get; set; }
+        [BindProperty(SupportsGet = true)] public bool LogIn { get; set; } = false;
+        [BindProperty(SupportsGet = true)] public bool LogOut { get; set; } = false;
+
+        public async Task OnGetAsync()
+        {
+            if (LogIn)
+            {
+                var prop = new AuthenticationProperties {RedirectUri = RedirectUri};
+                string auth = HttpContext.Request.Headers["Authorization"];
```

Comment:
The documentation that Microsoft has would have the login and logout schema executed in a separate page... Something about how it renders authorization requires that a challenge be done within the OnGet(), So normally that would be done by href to a new page(This is what the login/logout pages were doing in the previous commit). I minimized it to one page here by doing a query style submit instead of a route.

If you don't think that this is the way to go I can revert.

Reply:
That is a string that is never used anywhere. Good catch.